### PR TITLE
fix(HMSPROV-377) - disable image's launch button when image is not ready

### PR DESCRIPTION
**Pull request by amirfefer**
before: 
Screen Shot 2022-12-01 at 18 44 46

after:
Screen Shot 2022-12-01 at 18 42 51

### CODE REVIEW

File: src/Components/ImagesTable/ImageLink.js [hunk 1]
```diff
@@ -26,20 +26,25 @@ const ProvisioningLink = ({ imageId, isExpired, isInClonesTable }) => {
   );
 
   if (!error) {
-    image = image.isClone ? parent : image;
+    const provisionedImage = image.isClone ? parent : image;
     return (
       <Suspense fallback="loading">
-        <Button variant="link" isInline onClick={() => openWizard(true)}>
+        <Button
+          isDisabled={image.status !== 'success'}
```
Comment (croissanne) [inline src/Components/ImagesTable/ImageLink.js:33]: Can we move this to/solve this in ImageLink directly? It currently returns null if there's no uploadstatus, but might as well solve it for all cases, and require that the uploadstatus is defined and has succeeded.

**Review by croissanne — CHANGES_REQUESTED** [review]
One comment, otherwise lgtm.

Currently we don't disable the button, we just omit it in case it's not ready. Let's just omit it in case of failure as well.

File: src/Components/ImagesTable/ImageLink.js [hunk 2]
```diff
@@ -26,20 +26,25 @@ const ProvisioningLink = ({ imageId, isExpired, isInClonesTable }) => {
   );
 
   if (!error) {
-    image = image.isClone ? parent : image;
+    const provisionedImage = image.isClone ? parent : image;
```
Comment (lucasgarfield) [inline src/Components/ImagesTable/ImageLink.js:29]: This is a super nit-pick but now that we don't re-assign `image` can we make it a `const` (on line 11)?

**Comment by ezr-ondrej** [comment]
Implementing in #954 it was a one liner :)

**Comment by ezr-ondrej** [comment]
#954 is in, so I guess we can close this one @lucasgarfield ? :)

**Pull request closed without merging**